craft_application/models/project.py

```diff
@@ -67,3 +78,9 @@ def effective_base(self) -> Any:  # noqa: ANN401 app specific classes can improv
         if self.base is not None:
             return self.base
         raise RuntimeError("Could not determine effective base")
+
+    def get_build_plan(self) -> List[BuildInfo]:
+        """Obtain the list of architectures and bases from the project file."""
+        raise NotImplementedError(
```

Comment:
It was previously set as `abstractmethod` however it was quite inconvenient to make linters happy and it's useful to instantiate this class for testing when the method is not called. Also if later we implement the Grand Unified Platform Definition it can be handled by an instance of this class without changing it back from abstract to concrete. Note that this error will show up at application development time and won't affect application end users.